cmd/zdb/zdb.c

```diff
@@ -6335,6 +6335,33 @@ zdb_vdev_lookup(vdev_t *vdev, const char *path)
 	return (NULL);
 }
 
+static int
+name_from_objset_id(spa_t *spa, char *instr, char *outstr)
+{
+	uint64_t objset_id;
+	dsl_dataset_t *ds;
+
+	objset_id = strtoull(instr, NULL, 0);
```

Comment:
Setting a non-NULL `endstr` and checking it would be the traditional way to determine how the return value should be interpreted.